test/unit/specs/core/executeCommandFactory.spec.js

```diff
@@ -101,6 +101,30 @@ describe("executeCommandFactory", () => {
     });
   });
 
+  it("executes component commands", () => {
+    const runCommandSpy = jasmine.createSpy();
+    const testCommand = {
+      run: runCommandSpy,
+      optionsValidator: () => "options"
+    };
+    const componentRegistry = {
+      getCommand: () => testCommand,
+      getCommandNames() {
+        return ["test"];
+      }
+    };
+    const configureCommand = () => Promise.resolve(componentRegistry);
+    const executeCommand = executeCommandFactory({
+      logger,
+      configureCommand,
+      handleError
+    });
+    executeCommand("configure");
+    return executeCommand("test").then(() => {
+      expect(runCommandSpy).toHaveBeenCalledWith("options");
+    });
```

Comment:
Hey, sorry to keep asking for changes, but I noticed we're skipping a fairly significant part here. `validateCommandOptions` should be injected into `executeCommandFactory` instead of imported. Then, `validateCommandOptions` should be mocked. Then the test should assert that:

1.  `validateCommandOptions` was called with `command` and `options`.
2.  `runCommandSpy` has been called with whatever the mock `validateCommandOptions` returns.

Without this, there's no way of verifying that:

1. `validateCommandOptions` is being called and that it's receiving what it's expecting.
1. We're passing the results from `validateCommandOptions` to the command (which is an important part of ensuring the command will receive options with their default values set).